### PR TITLE
Add "Cleanup entry" in the entryeditor

### PR DESCRIPTION
**Pull request by YANG-XingYu**
Add "Cleanup entry" in the entryeditor.

2023-02-18_19-00-36_1802x878

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Thanks for your interest in JabRef. Is there any issue or any other reason why you did this? 
In addition, please check your checkstyle

**Comment by YANG-XingYu** [comment]
I did not find any related issues.
In my personal experience, usually after fetching an entry and downloading the corresponding paper, I will add some info to the entry by using the entryeditor, then generate the key and cleanup the entry which moves the paper file from the "Download" folder to my "Library" folder.
Therefore it is very convenient if there are both "Generate key" and "Cleanup entry" buttons in the entryeditor.
In addition, since there are "Generate key" and "Cleanup entry" buttons together in the toolbar, it makes sense that the entryeditor has such two buttons together instead of just one.

**Comment by Siedlerchr** [comment]
From my side it looks good! Can you please add a changelog entry as well?

File: src/main/java/org/jabref/gui/cleanup/CleanupSingleAction.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        this.dialogService = dialogService;
+        this.stateManager = stateManager;
+        this.undoManager = undoManager;
+
+        this.executable.bind(ActionHelper.needsEntriesSelected(stateManager));
+    }
+
+    @Override
+    public void execute() {
+        isCanceled = false;
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/cleanup/CleanupSingleAction.java:45]: Although the executable binding is set, the executability of the command is not dependend on it. The executable property is just for the ui control to enable or disable it. An unknowing developer could still create the command in the code and execute it. So I believe an additional check for the requirements to be met is needed at the beginning of the executable command.

**Review by calixtus — COMMENTED** [review]
Everything else looks also good to me. Please add the additional check and the changelog entry and we'll merge. Thanks for your contribution!

File: src/main/java/org/jabref/gui/cleanup/CleanupSingleAction.java (hunk 1 above)
Comment (YANG-XingYu) [reply src/main/java/org/jabref/gui/cleanup/CleanupSingleAction.java:45]: In the file `CleanupAction.java`, there are additional checks at the beginning of the executable command:
```
if (stateManager.getActiveDatabase().isEmpty()) {
            return;
        }

if (stateManager.getSelectedEntries().isEmpty()) { // None selected. Inform the user to select entries first.
            dialogService.showInformationDialogAndWait(Localization.lang("Cleanup entry"), Localization.lang("First select entries to clean up."));
            return;
        }
```
However, it seems that these requirements are always met for an entry based cleanup once the entryeditor is opened.
What kinds of additional checks should I add?
Could you please give more details?

Comment (calixtus) [reply src/main/java/org/jabref/gui/cleanup/CleanupSingleAction.java:45]: Ooops, missed that. Ok, then it looks good for me

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**